pinning-services/index.js

```diff
@@ -210,26 +275,73 @@ function updateDnsRecordDigitalOcean(domain, recordType, recordName, recordData,
   console.log(`GET ${urlGet}`);
   return fetch(urlGet, { headers: headers })
     .then(res => res.json())
-    .then(json => json['domain_records'].filter(record => record['type'] === recordType && record['name'] === recordName))
+    .then(json => {
+      if (json['id'] && json['id'] === 'not_found') {
+        return addDomainAccountDigitalOcean(domain, doToken)
+          .then(res => {
+            if (res) {
+              return [];
+            } else {
+              throw `Could not create domain ${domain}`;
+            }
+          });
+      } else {
+        return json['domain_records'].filter(record => record['type'] === recordType && record['name'] === recordName);
+      }
+    })
     .then(txt => {
+      let isDelete = false;
+      let isUpdate = false;
+
+      // If there is anything other then one entry found, 
+      // all entries should be removed then one entry added
+      if (txt.length != 1)
+        isUpdate = true
+
+      // If there is a record, and recordData is empty an string, delete only
```

Comment:
Huh? What is this case?

Reply:
passing "" as the data deletes the record.

So if a record exists and your passing "" as data, delete it (don't change the data to "")

Reply:
Ah right I figured out afterwards tks.